api/app/main.py

```diff
@@ -54,11 +55,14 @@ def create_app():
     app.include_router(users.router)
     app.include_router(threat.router)
 
-    # setup firebase
-    cred = setup_firebase_auth()
+    # setup auth
+    auth_module = FirebaseAuthModule()
+
+    def override_get_auth_module():
+        return auth_module
 
     # Dependency injection as needed
-    app.dependency_overrides[get_firebase_credentials] = lambda: cred
+    app.dependency_overrides[get_auth_module] = override_get_auth_module
```

Comment:
以下でもいいかもしれません (わかりにくければ元の方でもいいかなと)
```suggestion
    app.dependency_overrides[get_auth_module] = lambda: auth_module
```